lectures/inventory_dynamics.md

```diff
@@ -375,26 +372,23 @@ def compute_freq(firm, key,
     Xs = jnp.vstack((X, restock_count))
 
     # Define the function for each update
-    def update_X(Xs, D):
-
+    def update_X(i, Xs):
         # Separate the inventory and restock counter
-        X = Xs[0]
-        restock_count =  Xs[1]
-
-        restock_count = jnp.where(X <= s,
-                            restock_count + 1,
-                            restock_count)
-        X = jnp.where(X <= s, 
-                      jnp.maximum(S - D, 0),
-                      jnp.maximum(X - D, 0))
-        
-        Xs = jnp.vstack((X, restock_count))
-        return Xs, None
+        x, restock_cnt = Xs[0], Xs[1]
+        restock_cnt = jnp.where(x <= s,
+                                restock_cnt + 1,
+                                restock_cnt)
+        x = jnp.where(x <= s,
+                      jnp.maximum(S - D[i], 0),
+                      jnp.maximum(x - D[i], 0))
 
-    # Use lax.scan to perform the calculations on all states
-    X_final, _ = lax.scan(update_X, Xs, D)
+        Xs = jnp.vstack((x, restock_cnt))
```

Comment:
This is a comment about the existing code by @HumphreyYang , but why are we using `vstack` here?  Can't we just use a tuple?